### PR TITLE
web: allow custom host/port, default to 127.0.0.1 for security reasons 

### PR DESCRIPTION
**Pull request by cgodley**
1. Allow specifying the `--host` and `--port` for the web server to listen on
2. Default host to `127.0.0.1` for security reasons

### CODE REVIEW

**Comment by cgodley** [comment]
Hmm only 1 file should have changed, let me see what happened....

**Comment by cgodley** [comment]
Okay should be good to merge now. Sorry about that, still learning how to use GitHub

File: scripts/dream.py [hunk 1]
```diff
@@ -279,9 +279,9 @@ def dream_server_loop(t2i):
 
     # Start server
     DreamServer.model = t2i
-    dream_server = ThreadingDreamServer(("0.0.0.0", 9090))
+    dream_server = ThreadingDreamServer((host, port))
     print("\nStarted Stable Diffusion dream server!")
-    print("Point your browser at http://localhost:9090 or use the host's DNS name or IP address.")
+    print(f"Point your browser at http://{host}:{port} or use the host's DNS name or IP address.")
```
Comment (bakkot) [inline scripts/dream.py:284]:
```suggestion
    if host == '0.0.0.0':
      print(f"Point your browser at http://localhost:{port} or use the host's DNS name or IP address.")
    else:
      print(f"Point your browser at http://{host}:{port}.")
```

File: scripts/dream.py [hunk 2]
```diff
@@ -417,6 +417,18 @@ def create_argv_parser():
         action='store_true',
         help='Start in web server mode.',
     )
+    parser.add_argument(
+        '--host',
+        type=str,
+        default='127.0.0.1',
+        help='Web server: Host or IP to listen on'
```
Comment (bakkot) [inline scripts/dream.py:424]:
```suggestion
        help='Web server: Host or IP to listen on. Set to 0.0.0.0 to accept traffic other devices on your network.'
```

**Review by bakkot — COMMENTED** [review]
Thanks! I'm always a little bit wary of changing the defaults in a way which will make things which used to work stop working, so I'd like to hear opinions from other people about changing the default is a change worth making. But the change to support custom ports is certainly worth it.

**Comment by lstein** [comment]
I'm OK with changing the default behavior as long as we give the user a big fat warning that this has happened. I'm reviewing now, and will add a suitable warning message if the user does not specify --host.

**Review by lstein — APPROVED** [review]
Looks good.

**Pull request merged**